src/materials/Material.js

```diff
@@ -86,6 +86,15 @@ Object.assign( Material.prototype, EventDispatcher.prototype, {
 
 			}
 
+			// for backward compatability if shading is set in the constructor
+			if ( key === 'shading' ) {
+
+				console.warn( 'THREE.' + this.type + ': .shading has been removed. Use the boolean .flatShading instead.' );
+				this.flatShading = ( newValue === FlatShading ) ? true : false;
```

Comment:
I don't think it's that hacky... One more reason to not be a big fan of this properties object in constructors approach though.